### PR TITLE
Build script improvements

### PR DESCRIPTION
**Pull request by sraaphorst**
This PR comprises major changes to the way that the build script works.

Smaller changes include:
1. Scripts can now be run anywhere instead of requiring the user to be in the ocs-build dir.
2. Lots of cleanup.
3. Java version is now set (OS X only: will fail on other platforms, but not terminate build).

Larger changes include:
1. Previously, if SSH IDs expired, the scripts would prompt a password, and this would generally cause timeout and failure. Now, SSH IDs are always set up at the beginning of the script.
2. Functionality to continue the build process has now been implemented, so if there is a failure, the build can recommence where it left off instead of having to start over from scratch.

This took longer and was more elaborate than expected between the remembering bash programming, writing the code, debugging, and testing. 😄

### CODE REVIEW

File: common.sh [hunk 1]
```diff
@@ -15,24 +17,59 @@ OCS_BASE_PATH=`absPath ${OCS_BASE_PATH:-"$DEV_BASE_PATH/ocs"}`
...
+	exit 1
+    fi
+    logInfo "SSH ID setup complete."
+    return 0
+}
+
+# setjdk always returns 0, even for failure.
+function javaVersion() {
+    logHeader "Setting JDK version"
+    setjdk 1.8.0_73
```
Comment (cquiroz) [inline common.sh:48]: So we need to remember to update this when we upgrade the jvm?

File: jdkfuncs.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# This code taken from:
+# https://www.jayway.com/2014/01/15/how-to-switch-jdk-version-on-mac-os-x-maverick
+# It allows setting the JDK version to use in the current process.
+function setjdk() {
```
Comment (cquiroz) [inline jdkfuncs.sh:5]: Just a note, I use [jEnv](http://www.jenv.be/) to change my jvm for different projects. You don't need to use it, is just a side note

**Comment by cquiroz** [comment]
You probably deserve a medal for this. What's the issue with ssh? At least my ssh keys never expire

File: jdkfuncs.sh (hunk 2 above)
Comment (sraaphorst) [inline jdkfuncs.sh:5]: Oh nice... that would work better for portability. I'll take a look! Thanks for the recommendation!

File: common.sh (hunk 1 above)
Comment (sraaphorst) [inline common.sh:48]: Yes, we will have to update this, unfortunately. I am hoping the message in the build feedback will at least remind us that it needs to be done, but am open to better suggestions.

**Comment by sraaphorst** [comment]
Thanks! I really appreciate the positive feedback!
Perhaps I am working with `ssh` differently than you or there is a setting that lets keys expire?
Are you running `ssh-copy-id` to set things up to avoid password entry?

**Comment by cquiroz** [comment]
I read a bit about `ssh-copy-id` and it does exactly what I do, but I just do it manually
The keys are for user `telops` right? I'm guessing the problem is tha ITS resets the telops user every now and then. Perhaps we can ask them to set it permanently

**Comment by swalker2m** [comment]
I skimmed through these changes and agree with Carlos, you probably deserve a medal.  I'm not going to go through it with a fine toothed comb.  If there are problems they should fall out of the test deploys we will do in November if not before.  👍

**Comment by sraaphorst** [comment]
Going to merge this for now, and look into switching to Carlos' JDK setting app possibly in the future, although I'm not convinced that we need it: seems that the last ODB build was accidentally built with 1.8.0_92, but packaged with 1.8.0_73 (which happens automatically due to the files in the `/Users/software/dev/jres8` directory), and the eObsLog issue that was coming up before was just fine running with the old JRE. (I guess this shouldn't be remotely surprising as we should expect backwards compatibility to that level in Java.)

**Pull request merged**